src/server/lib/user-features.ts

```diff
@@ -42,3 +60,47 @@ export const setUserFeatureCookies = async ({
 		});
 	}
 };
+
+export const createCookie = ({
+	name,
+	res,
+	daysTillExpiry,
+}: {
+	name: string;
+	res: Response;
+	daysTillExpiry: number;
+}) => {
+	const tmpDate = new Date(Date.now() + daysTillExpiry * 24 * 60 * 60 * 1000);
+	res.cookie(name, tmpDate.getTime().toString(), {
+		domain,
+		expires: tmpDate,
+	});
+};
+
+export const persistUserBenefitsCookies = ({
+	userBenefits,
+	res,
+}: {
+	userBenefits: UserBenefitsResponse;
+	res: Response;
+}) => {
+	if (userBenefits?.benefits?.includes('hideSupportMessaging')) {
+		createCookie({
+			name: HIDE_SUPPORT_MESSAGING_COOKIE,
+			res,
+			daysTillExpiry: 7,
+		});
+	}
+	if (userBenefits?.benefits?.includes('allowRejectAll')) {
+		createCookie({ name: ALLOW_REJECT_ALL_COOKIE, res, daysTillExpiry: 7 });
+	}
+	if (userBenefits?.benefits?.includes('adFree')) {
+		createCookie({ name: AD_FREE_USER_COOKIE, res, daysTillExpiry: 2 });
+	}
```

Comment:
For each of these benefits, can we add a comment that link to the code in the DCR repo, again like we did previosuly for members-data-api:

https://github.com/guardian/gateway/blob/34311cdf8524f247a11841fbf7bdf9568f5829ea/src/server/lib/user-features.ts#L34-L42